os/arch/arm/src/common/up_checkspace.c

```diff
@@ -38,11 +38,23 @@ extern uint32_t _sdata;
 extern uint32_t _edata;
 extern uint32_t _sbss;
 extern uint32_t _ebss;
+extern uint32_t _ram_kernel_text_start;
+extern uint32_t _ram_kernel_text_end;
+
+bool is_kernel_text_space(void *addr)
+{
+	/* Check if the address lies in the kernel text section range */
+	if ((addr >= (void *)&_stext && addr <= (void *)&_etext) ||
+		(addr >= (void *)&_ram_kernel_text_start && addr <= (void *)&_ram_kernel_text_end)) {
```

Comment:
Are `_ram_kernel_text_start` and `_ram_kernel_text_end` working even there is no ram text space?